File: Algorithms/C++/TwoSum/TwoSum.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int, int> numbers;

        for(int i = 0; i < nums.size(); i++){
            numbers[nums[i]] = i;
        }

        for(int i = 0; i < nums.size(); i++){
            int tempnum = target - nums[i];
            // find the tempnum in the numbers and check it is not the same index
            if(numbers.count(tempnum) && numbers[tempnum] != i){
                return {i, numbers[tempnum]};
            }
        }

        return {};
    }
};

File: Algorithms/C#/1.TwoSum/two_sum.cs
using System;

public class Solution
{

    int i = 0;

    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> temp = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            int left = target - nums[i];
            if (temp.ContainsKey(left))
            {
                return new int[] { temp[left], i };
            }
            if (!temp.ContainsKey(nums[i]))
            {
                temp.Add(nums[i], i);
            }
        }
        return null;
    }

    public Main()
    {
        int[] nums = new int[2, 7, 11, 15];

        target = 9;

        TwoSum(nums, target);

        console.writeline("[", i, ",", i + 1, "]");
    }

}
